### PR TITLE
Add GPU clusters to deployments

### PR DESCRIPTION
**Pull request by will-cromar**
# Description

Add GPU cluster to deployment config.

# Tests

Applied config. Tested in PR that adds the tests: https://github.com/GoogleCloudPlatform/ml-auto-solutions/pull/111#issuecomment-1949326793

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run one-shot tests and provided workload links above if applicable. 
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

**Comment by will-cromar** [comment]
Pulling this back to draft. I need to figure out how to deploy the headless service (for service discovery) from terraform.

**Review by RissyRan — COMMENTED** [review]
Thanks Will! I don't have comment about Terraform template. Will those GPU capacity is taken once cluster is up? If so, @zpcore could you check if this change will affect your team's tests? I understand the resource request is painful.

**Comment by will-cromar** [comment]
I've had a cluster of a the same size up for a few days for testing. Let me know if that caused a capacity crunch. I can switch this to a more conservative auto-scaling policy if needed.

**Comment by zpcore** [comment]
Thanks for the head up. The GPU torchbench tests are scheduled to run at 3:00am PT. We will see tomorrow.

File: deployment/clusters.tf [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  host  = "https://${google_container_cluster.gpu-uc1.endpoint}"
+  token = data.google_client_config.provider.access_token
+  cluster_ca_certificate = base64decode(
+    google_container_cluster.gpu-uc1.master_auth[0].cluster_ca_certificate,
+  )
+}
+
+// Headless service required for service discovery within a Job.
+// Pods will be addressable as `hostname.headless-svc`
+resource "kubernetes_service" "example" {
```
Comment (vanbasten23) [inline deployment/clusters.tf:85]: Does it mean it is a static and long lived service that we only run once?

Comment (will-cromar) [reply deployment/clusters.tf:85]: Right. Every pod that defines the label `headless-svc=true` will register under the same service. 

I would have liked to create a service for each job that matches the first worker, but confusingly the job index is defined as an annotation and not a label. Services can only match labels.

Comment (vanbasten23) [reply deployment/clusters.tf:85]: Got it. Each job has a unique name and they all share the same service (name is static).

**Pull request merged**